src/transformers/models/donut/processing_donut.py

```diff
@@ -85,6 +91,14 @@ def __call__(
         [`~DonutTokenizer.__call__`]. Please refer to the doctsring of the above two methods for more information.
         """
         # For backward compatibility
+        legacy = kwargs.pop("legacy", True)
+        if legacy:
+            logger.warning_once(
+                "Legacy behavior is being used. The new behavior with legacy=False will be enabled in the future."
+                "In the new behavior, if both images and text are provided, the default value of `add_special_tokens` "
+                "will be changed to `False` when calling the tokenizer if `add_special_tokens` is unset."
+            )
```

Comment:
can you add a comment on why we need to have `add_special_tokens` to help future ourselves when reading the code?